pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/FixedByteChunkSingleValueReader.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.io.reader.impl.v1;
+
+import com.linkedin.pinot.core.io.compression.ChunkDecompressor;
+import com.linkedin.pinot.core.io.reader.impl.ChunkReaderContext;
+import com.linkedin.pinot.core.io.writer.impl.v1.FixedByteChunkSingleValueWriter;
+import com.linkedin.pinot.core.segment.memory.PinotDataBuffer;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+
+/**
+ * Reader class for data written out by {@link FixedByteChunkSingleValueWriter}.
+ * For data layout, please refer to the documentation for {@link FixedByteChunkSingleValueReader}
+ *
+ */
+public class FixedByteChunkSingleValueReader extends BaseChunkSingleValueReader {
+
+  /**
+   * Constructor for the class.
+   *
+   * @param pinotDataBuffer Data buffer to read from
+   * @param uncompressor Chunk uncompressor
+   * @throws IOException
+   */
+  public FixedByteChunkSingleValueReader(PinotDataBuffer pinotDataBuffer, ChunkDecompressor uncompressor)
+      throws IOException {
+    super(pinotDataBuffer, uncompressor);
+  }
+
+  @Override
+  public int getInt(int row, ChunkReaderContext context) {
+    assert _lengthOfLongestEntry == INT_SIZE;
```

Comment:
I thought about that as well. The reason I put in an assert is for readibility. As asserts are turned off in production, it shouldn't cause an issue. We have similar asserts elsewhere in the code (ie per row).